tests/Unit/Domain/CoordinateMaps/Test_CoordinateMap.cpp

```diff
@@ -902,14 +902,13 @@ void test_time_dependent_map() {
 
   const std::array<DataVector, deriv_order + 1> init_func{
       {{1.0}, {-2.0}, {2.0}, {0.0}}};
-  domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>
-      function_of_time_derived(initial_time, init_func);
-  domain::FunctionsOfTime::FunctionOfTime& function_of_time =
-      function_of_time_derived;
-
-  const std::unordered_map<std::string,
-                           domain::FunctionsOfTime::FunctionOfTime&>
-      functions_of_time = {{"trans", function_of_time}};
+  using Polynomial = domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>;
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      functions_of_time{};
+  functions_of_time["trans"] =
```

Comment:
Does the initializer list not work for this one? it would be preferable to const with initializer list if it works. (same question for a few other tests below)

Reply:
It does not, or I could not convince both clang and GCC to agree on the syntax. There's some internal copy going on in the implementation of unordered_map (I'm not sure whether that's a bug in the STL or not)

Reply:
Initializer lists are always const, so they're questionable-to-broken with anything with a non-trivial (or missing) copy operation.

My understanding is that this is because they were developed concurrently with move semantics so didn't consider them in the design.  I've never heard a reason why the standards committee hasn't fixed this in subsequent versions.